crates/autopilot/src/infra/solvers/authenticator.rs

```diff
@@ -0,0 +1,107 @@
+use {
+    crate::{
+        domain::{self, eth},
+        infra::blockchain::{contracts::deployment_address, ChainId},
+    },
+    ethcontract::dyns::DynWeb3,
+    primitive_types::H160,
+};
+
+#[allow(dead_code)]
+#[derive(Debug, Clone)]
+pub struct Authenticator {
+    /// The authenticator contract used for allow-listing solvers to settle.
+    authenticator: contracts::GPv2AllowListAuthentication,
+    /// The safe module that is used to provide special role to EOA.
+    authenticator_role: contracts::Roles,
+    /// The EOA that is allowed to add/remove solvers.
```

Comment:
For the purposes of the circuit breaker adding solvers is not important so I wouldn't mention it.